code/ai/aicode.cpp

```diff
@@ -13677,8 +13683,11 @@ void ai_execute_behavior(ai_info *aip)
 		// FIXME: got this from TBP and added it here to skip the Int3() but don't really want to handle it
 		// properly until after 3.6.7 just to avoid delaying release or breaking something - taylor
 		break;
+	case AIM_LUA:
+		ai_lua(&Ships[Pl_objp->instance]);
```

Comment:
If you want to make Lua AI applicable to anything in the OSWPT package, then I don't think `&Ships[Pl_objp->instance]` is appropriate here

Reply:
Pl_objp is in fact the ship that will be lua-controlled and not its target, but you're very right that it's not clean to directly pass it here. It can easily be the aip pointer itself.
Changed.

Reply:
Ah right